src/qibolab/instruments/simulator/backends/generic.py

```diff
@@ -0,0 +1,284 @@
+from collections import OrderedDict
+from typing import List, Optional, Union
+
+import numpy as np
+from IPython.display import Latex, display
+
+
+def dec_to_basis_string(
+    x: int, nlevels: list = [2], to_string: bool = True
+) -> Union[str, list]:
+    """Converts an integer to a generalized bitstring in the computation basis
+    of the full Hilbert space.
+
+    Args:
+        x (int): The integer (decimal number) to be converted.
+        nlevels (list): The list of nlevels to convert the decimal number to. Defaults to [2].
+        to_string (bool): Flag to return the result as a string. Defaults to True.
+
+    Returns:
+        str or list: Generalized bitstring of x. If to_string is True, the result is returned as a string, otherwise as a list of integers.
+    """
+
+    nqubits = len(nlevels)
+
+    output_list = []
+    y = x
+    for i in range(nqubits):
+        if i < nqubits - 1:
+            sub_dim = np.prod(nlevels[i + 1 :])
+            coeff = np.divmod(y, sub_dim)[0]
+            output_list.append(coeff)
+            y -= coeff * sub_dim
+        else:
+            coeff = np.divmod(y, 1)[0]
+            output_list.append(coeff)
+
+    if to_string:
+        output_list = "".join([str(c) for c in output_list])
```

Comment:
Avoid this:
- it is completely decoupled, so it can happen in a further step
- if you want to provide something like this, make a separate function
- you will simplify the signature (one parameter less) and improve composition

Reply:
Removed to_string flag in 93eb292 (and fixed related sections that calls this function in 6bfe896 and 6f12cff)